units/cache_historical_test_data.go

```diff
@@ -0,0 +1,289 @@
+	return j
+}
+
+func makeCacheHistoricalTestDataJob() *cacheHistoricalTestDataJob {
+	j := &cacheHistoricalTestDataJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    cacheHistoricalTestDataName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+	return j
+}
+
+func (j *cacheHistoricalTestDataJob) Run(ctx context.Context) {
+	defer j.MarkComplete()
+
+	projectId := j.ProjectId
+
+	grip.Info(message.Fields{
+		"job_id":     j.ID(),
+		"project_id": projectId,
+		"message":    "starting job",
+	})
+
+	// Lookup last sync date for project
+	statsStatus, err := stats.GetStatsStatus(projectId)
+	if err != nil {
+		if err != nil {
+			j.AddError(errors.Wrap(err, "error retrieving last sync date"))
+			return
+		}
+	}
+
+	tasksToIgnore, err := getTasksToIgnore(j.ProjectId)
+	if err != nil {
+		if err != nil {
+			j.AddError(errors.Wrap(err, "error retrieving project settings"))
+			return
+		}
+	}
+
+	syncFromTime := statsStatus.ProcessedTasksUntil
+	syncToTime := findTargetTimeForSync(syncFromTime)
+	jobTime := time.Now()
+
+	grip.Info(message.Fields{
+		"job_id":  j.ID(),
+		"message": fmt.Sprintf("Syncing data between %v - %v", syncFromTime, syncToTime),
+	})
+
+	statsToUpdate, err := stats.FindStatsToUpdate(projectId, syncFromTime, syncToTime)
+	if err != nil {
+		j.AddError(errors.Wrap(err, "error finding tasks to update"))
+		return
+	}
+
+	generateMap := generateFunctions{
+		HourlyFns: map[string]generateStatsFn{
+			"test": stats.GenerateHourlyTestStats,
+		},
+		DailyFns: map[string]generateStatsFn{
+			"test": stats.GenerateDailyTestStatsFromHourly,
+			"task": stats.GenerateDailyTaskStats,
+		},
+	}
+
+	err = updateHourlyAndDailyStats(projectId, statsToUpdate, jobTime, generateMap, tasksToIgnore)
+	if err != nil {
+		j.AddError(errors.Wrap(err, "error generating hourly test stats"))
+		return
+	}
+
+	// update last sync
+	err = stats.UpdateStatsStatus(projectId, jobTime, syncToTime)
+	if err != nil {
+		j.AddError(errors.Wrap(err, "error updating last synced date"))
+		return
+	}
+
+	grip.Info(message.Fields{
+		"job_id":     j.ID(),
+		"project_id": projectId,
+		"message":    "job completed",
+	})
+}
+
+func getTasksToIgnore(projectId string) ([]*regexp.Regexp, error) {
+	ref, err := model.FindOneProjectRef(projectId)
+	if err != nil {
+		return nil, err
+	}
+
+	filePatternsStr := ref.FilesIgnoredFromCache.FilePatterns
+
+	return splitPatternStringToRegexList(filePatternsStr)
+}
+
+func splitPatternStringToRegexList(filePatternsStr string) ([]*regexp.Regexp, error) {
+	filePatterns := strings.Split(filePatternsStr, ",")
+
+	var tasksToIgnore []*regexp.Regexp
+	for _, patternStr := range filePatterns {
+		pattern := strings.Trim(patternStr, " ")
+		if pattern != "" {
+			regexp, err := regexp.Compile(pattern)
+			if err != nil {
+				grip.Warning(message.Fields{
+					"message": fmt.Sprintf("Could not compile regexp from '%s'", pattern),
+				})
```

Comment:
there's a grip.Warningf() which has the same sigature as message.Fields. only use message.fields if you want to pass structured data to the logger.

Reply:
changed.